lib/supply/client.rb

```diff
@@ -18,38 +19,27 @@ class Client
     # @!group Login
     #####################################################
 
-    # Initializes the auth_client and api_client using the specified information
-    # @param path_to_key: The path to your p12 file
-    # @param issuer: Email addresss for oauth
-    # @param passphrase: Passphrase for the p12 file
-    def initialize(path_to_key: nil, issuer: nil, passphrase: nil)
-      passphrase ||= "notasecret"
+    # Initializes the android_publisher and its auth_client using the specified information
+    # If not specified, it tries to take the value of the secrets.json from ENV['GOOGLE_APPLICATION_CREDENTIALS']
+    # @param path_to_service_account_json: The path to your service account Json file
+    def initialize(path_to_service_account_json: nil)
+      # REVIEW: do we want to force override ? If not, we don't get the 'proper'
+      # printed table output
+      ENV['GOOGLE_APPLICATION_CREDENTIALS'] = File.expand_path path_to_service_account_json if path_to_service_account_json
 
-      key = Google::APIClient::KeyUtils.load_from_pkcs12(File.expand_path(path_to_key), passphrase)
+      scopes = [Androidpublisher::AUTH_ANDROIDPUBLISHER]
 
-      begin
-        self.auth_client = Signet::OAuth2::Client.new(
```

Comment:
To implement that I would be depending on the google auth library to still maintain this functionality, something they don't even document} and in fact is not recommended:

![image](https://cloud.githubusercontent.com/assets/24282/12946786/47fc1e1c-cff7-11e5-9d4b-022a786bdf70.png)

As I depend on google auth to provide this feature, I have no guarantee that this will work in future releases of google auth, so it's risky in my own opinion.

Having had to suffer multiple breaking changes in fastlane tools pre 1.0.0, I would prefer to use my energy to add other missing features than supporting a deprecated api. For example, right now I cannot use this lib until it supports #42 which is next in my task list :)